### PR TITLE
gh-104711: Add security warning to the CGIHTTPRequestHandler document

### PR DESCRIPTION
**Pull request by aisk**
Also related to #104712. cc @zooba 

----
📚 Documentation preview 📚: https://cpython-previews--115915.org.readthedocs.build/

* Issue: gh-104711

### CODE REVIEW

**Comment by zooba** [comment]
It's a starting point, but let's focus on "only use for trusted clients". We're not going to remove it, we just don't want people expecting that it's safe to let people they don't trust access it over the internet.

**Comment by zooba** [comment]
Oh I just noticed the deprecation/pending removal message directly above in the file, so I guess we are planning to remove it. There probably isn't any need to add an extra warning in that case?

I don't know where the deprecation was decided, or if it was for security reasons. If so, we could just mention in the deprecation message that this should not be used for public-facing services and not add another warning. Otherwise, maybe the security section that immediately follows is a better place to mention it (and easier to backport, too).

File: Doc/library/http.server.rst [hunk 1]
```diff
@@ -520,6 +520,12 @@ the ``--cgi`` option::
    :mod:`http.server` command line ``--cgi`` support is being removed
    because :class:`CGIHTTPRequestHandler` is being removed.
 
+.. warning::
+
+   :class:`CGIHTTPRequestHandler` and the ``--cgi`` command line option
+   have known security issues, therefore you should not expose them
+   to untrusted clients. Use it at your own risk.
```
Comment (zooba) [inline Doc/library/http.server.rst]:
```suggestion
   are not intended for use by untrusted clients and may be vulnerable
   to exploitation. Always use within a secure environment.
```

How about this? I think it's a little less down on ourselves, and hopefully gives users something they can actually do (i.e. "I should secure my environment")

**Comment by zooba** [comment]
LGTM, let me know when you're happy to merge.

**Comment by aisk** [comment]
Thanks for the review! I think this change is ready to merge.

**Comment by aisk** [comment]
Sorry, as you pointed out, these changes introduced a conflict during the backport 😂. I resolved it manually.

**Comment by zooba** [comment]
Thanks! Merged now

**Pull request merged**